torch_xla/csrc/ops/select.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "ir.h"
+#include "tensorflow/compiler/xla/xla_data.pb.h"
+#include "tensorflow/core/lib/gtl/array_slice.h"
+
+namespace torch_xla {
+namespace ir {
+namespace ops {
+
+class Select : public Node {
+ public:
+  Select(const Value& input, xla::int64 dim, xla::int64 index);
+
+  XlaOpVector Lower(LoweringContext* loctx) const override;
+
+  std::string ToString() const override;
+
+  xla::int64 dim() const { return dim_; }
+
+  xla::int64 index() const { return index_; }
+
+ private:
+  xla::int64 dim_ = 0;
```

Comment:
Why are you initializing these fields here? It's not needed.

Reply:
In this case, no (there is no other way to construct aside from the single constructor).
It is something which I get used to and prevents surprises (as in, random behavior) in case you miss initialization.

Reply:
Dropped.